httpx/utils.py

```diff
@@ -144,6 +145,21 @@ def parse_header_links(value: str) -> typing.List[typing.Dict[str, str]]:
     return links
 
 
+def get_environment_proxies() -> typing.Dict[str, str]:
+    """Gets proxy information from the environment"""
+
+    # urllib.request.getproxies() falls back on System
+    # Registry and Config for proxies on Windows and macOS.
+    proxies = {key: val for key, val in getproxies().items()}
+
+    # Favor lowercase environment variables over uppercase.
```

Comment:
Could we clarify we do this because `all_proxy` is by convention generally given as lowercase, contrary to most environment variables?